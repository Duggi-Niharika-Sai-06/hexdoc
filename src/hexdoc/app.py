from flask import Flask, render_template

app = Flask(__name__)

@app.route('/')
def index():
    return render_template('table_of_contents.html.jinja')

if __name__ == '__main__':
    app.run(debug=True)
