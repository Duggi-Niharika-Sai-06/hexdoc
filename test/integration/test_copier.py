# pyright: reportUnknownMemberType=false

import subprocess
import sys
from pathlib import Path

import pytest
from hexdoc.cli.app import render
from pytest import MonkeyPatch
from syrupy.assertion import SnapshotAssertion

from ..conftest import list_directory, write_file_tree


def run_pip(*args: str):
    return subprocess.run((sys.executable, "-m", "pip") + args, check=True)


@pytest.mark.copier
def test_copier(
    monkeypatch: MonkeyPatch,
    snapshot: SnapshotAssertion,
    path_snapshot: SnapshotAssertion,
    env_overrides: dict[str, str],
):
    monkeypatch.chdir("submodules/hexdoc-hexcasting-template/.ctt/test_copier")

    write_file_tree(
        ".",
        {
            "src/generated/resources": {},
            "src/main/java/com/package/Patterns.java": "",
            "src/main/resources/assets/hexcasting/patchouli_books/thehexbook/en_us": {
                "categories/foo.json": {
                    "name": "hexdoc.mod.title",
                    "icon": "minecraft:amethyst_shard",
                    "description": "hexcasting.category.basics.desc",
                    "sortnum": 0,
                },
                "entries/bar.json": {
                    "name": "hexdoc.welcome.header",
                    "category": "hexcasting:foo",
                    "icon": "minecraft:textures/mob_effect/nausea.png",
                    "sortnum": 0,
                    "advancement": "hexcasting:y_u_no_cast_angy",
                    "pages": [
                        {
                            "type": "patchouli:text",
                            "text": "hexcasting.page.couldnt_cast.1",
                        },
                    ],
                },
            },
            ".env": "\n".join(
                f"{name}={value}" for name, value in env_overrides.items()
            ),
            "gradle.properties": (
                """\
                modVersion=1.0.0
                hexcastingVersion=0.11.1-7
                minecraftVersion=1.20.1
                """
            ),
        },
    )

    output_dir = Path("_site")

    try:
        run_pip("install", ".", "--force-reinstall", "--no-deps")
        render(
            output_dir=output_dir,
            props_file=Path("doc/hexdoc.toml"),
            branch="main",
        )
    finally:
        run_pip("uninstall", "hexdoc-mod", "-y")

    path_snapshot._custom_index = "vlatestmainen_usindex.html"  # pyright: ignore[reportPrivateUsage]
    assert output_dir / "v/latest/main/en_us/index.html" == path_snapshot

    assert list_directory(output_dir) == snapshot
