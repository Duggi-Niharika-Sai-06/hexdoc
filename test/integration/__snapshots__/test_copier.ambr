# serializer version: 1
# name: test_copier
  list([
    'v',
    'v/latest',
    'v/latest/main',
    'v/latest/main/en_us',
    'v/latest/main/en_us/.sitemap-marker.json',
    'v/latest/main/en_us/apple-touch-icon-114x114.png',
    'v/latest/main/en_us/apple-touch-icon-120x120.png',
    'v/latest/main/en_us/apple-touch-icon-144x144.png',
    'v/latest/main/en_us/apple-touch-icon-152x152.png',
    'v/latest/main/en_us/apple-touch-icon-167x167.png',
    'v/latest/main/en_us/apple-touch-icon-180x180.png',
    'v/latest/main/en_us/apple-touch-icon-57x57.png',
    'v/latest/main/en_us/apple-touch-icon-60x60.png',
    'v/latest/main/en_us/apple-touch-icon-72x72.png',
    'v/latest/main/en_us/apple-touch-icon-76x76.png',
    'v/latest/main/en_us/favicon-16x16.png',
    'v/latest/main/en_us/favicon-180x180.png',
    'v/latest/main/en_us/favicon-196x196.png',
    'v/latest/main/en_us/favicon-32x32.png',
    'v/latest/main/en_us/favicon-64x64.png',
    'v/latest/main/en_us/favicon-96x96.png',
    'v/latest/main/en_us/favicon.ico',
    'v/latest/main/en_us/hexcasting.js',
    'v/latest/main/en_us/icon.png',
    'v/latest/main/en_us/index.css',
    'v/latest/main/en_us/index.html',
    'v/latest/main/en_us/index.js',
    'v/latest/main/en_us/mstile-270x270.png',
    'v/latest/main/en_us/mstile-310x150.png',
    'v/latest/main/en_us/mstile-310x310.png',
    'v/latest/main/en_us/mstile-70x70.png',
    'v/latest/main/en_us/textures.css',
  ])
# ---
